pypeit/spectrographs/mmt_binospec.py

```diff
@@ -1,959 +1,405 @@
-    # that forces a file to be provided?
-
-    def get_slitmask(self, filename):
-        hdu = fits.open(filename)
-        corners = np.array([hdu['BluSlits'].data['slitX1'],
-                            hdu['BluSlits'].data['slitY1'],
-                            hdu['BluSlits'].data['slitX2'],
-                            hdu['BluSlits'].data['slitY2'],
-                            hdu['BluSlits'].data['slitX3'],
-                            hdu['BluSlits'].data['slitY3'],
-                            hdu['BluSlits'].data['slitX4'],
-                            hdu['BluSlits'].data['slitY4']]).T.reshape(-1, 4, 2)
-        self.slitmask = SlitMask(corners, slitid=hdu['BluSlits'].data['dSlitId'])
-        return self.slitmask
-
-    def get_grating(self, filename):
         """
-        Taken from xidl/DEEP2/spec2d/pro/deimos_omodel.pro and
-        xidl/DEEP2/spec2d/pro/deimos_grating.pro
-        """
-        hdu = fits.open(filename)
-
-        # Grating slider
-        slider = hdu[0].header['GRATEPOS']
-        # TODO: Add test for slider
-
-        # Central wavelength, grating angle, and tilt position
-        if slider == 3:
-            central_wave = hdu[0].header['G3TLTWAV']
-            # Not used
-            # angle = (hdu[0].header['G3TLTRAW'] + 29094)/2500
-            tilt = hdu[0].header['G3TLTVAL']
-        elif slider in [2, 4]:
-            # Slider is 2 or 4
-            central_wave = hdu[0].header['G4TLTWAV']
-            # Not used
-            # angle = (hdu[0].header['G4TLTRAW'] + 40934)/2500
-            tilt = hdu[0].header['G4TLTVAL']
-        else:
-            raise ValueError('Slider has unknown value: {0}'.format(slider))
-
-        # Ruling
-        name = hdu[0].header['GRATENAM']
-        if 'Mirror' in name:
-            ruling = 0
-        else:
-            # Remove all non-numeric characters from the name and
-            # convert to a floating point number
-            ruling = float(re.sub('[^0-9]', '', name))
-            # Adjust
-            if abs(ruling - 1200) < 0.5:
-                ruling = 1200.06
-            elif abs(ruling - 831) < 2:
-                ruling = 831.90
-
-        # Get the orientation of the grating
-        roll, yaw, tilt = KeckDEIMOSSpectrograph._grating_orientation(slider, ruling, tilt)
-
-        self.grating = None if ruling == 0 else ReflectionGrating(ruling, tilt, roll, yaw,
-                                                                  central_wave=central_wave)
-        return self.grating
-
-    def get_detector_map(self):
-        if self.detector_map is None:
-            self.detector_map = DEIMOSDetectorMap()
-        return self.detector_map
-
-    @staticmethod
-    def _grating_orientation(slider, ruling, tilt):
-        """
-        Return the roll, yaw, and tilt of the grating.
+        msgs.info("Custom bad pixel mask for BINOSPEC")
+        bpm_img = self.empty_bpm(filename, det, shape=shape)
```

Comment:
If it is empty, you don't need this method.

Reply:
Removed.